src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp

```diff
@@ -47,634 +47,383 @@
 
 #ifdef KOKKOSKERNELS_ENABLE_TPL_MKL
 #include "mkl_spblas.h"
-#include "mkl.h"
 #endif
 
-#include "KokkosKernels_Utils.hpp"
-#include <Kokkos_Concepts.hpp>
-
 namespace KokkosSparse {
-
 namespace Impl {
 
-template <typename KernelHandle, typename in_row_index_view_type,
-          typename in_nonzero_index_view_type, typename bin_row_index_view_type,
-          typename bin_nonzero_index_view_type,
-          typename cin_row_index_view_type>
-void mkl_symbolic(KernelHandle *handle, typename KernelHandle::nnz_lno_t m,
-                  typename KernelHandle::nnz_lno_t n,
-                  typename KernelHandle::nnz_lno_t k,
-                  in_row_index_view_type row_mapA,
-                  in_nonzero_index_view_type entriesA,
-
-                  bool transposeA, bin_row_index_view_type row_mapB,
-                  bin_nonzero_index_view_type entriesB, bool transposeB,
-                  cin_row_index_view_type row_mapC, bool verbose = false) {
 #ifdef KOKKOSKERNELS_ENABLE_TPL_MKL
 
-  typedef typename KernelHandle::nnz_lno_t idx;
-  typedef typename KernelHandle::size_type size_type;
-
-  typedef typename KernelHandle::HandleTempMemorySpace HandleTempMemorySpace;
-  typedef typename Kokkos::View<int *, HandleTempMemorySpace>
-      int_temp_work_view_t;
+inline void mkl_call(sparse_status_t result, const char *err_msg) {
+  if (SPARSE_STATUS_SUCCESS != result) {
+    throw std::runtime_error(err_msg);
+  }
+}
 
-  typedef typename KernelHandle::nnz_scalar_t value_type;
+template <typename value_type>
+class MKLSparseMatrix {
```

Comment:
I do not disagree with the idea behind that class but I do not like that it is defined in `KokkosSparse_spgemm_mkl_impl.hpp`, I feel that we are bound to keep on re-implementing it in every kernel if we do that...

Reply:
Would it make sense to also define this class in `src/common/KokkosKernels_SparseUtils_mkl.hpp` to be shared between kernels ?

Reply:
I moved `MKLSparseMatrix` (without `mkl_spmm()`) to _src/common/KokkosKernels_SparseUtils_mkl.hpp_ on fefb5f406cb6013e4adac6c7a52e7f30631ef548